src/wasm.h

```diff
@@ -1994,11 +1994,13 @@ class ElementSegment : public Named {
 public:
   Name table;
   Expression* offset;
-  Type type = Type::funcref;
+  Type type = Type(HeapType::func, Nullable);
```

Comment:
Yes, default initializers are executed at object construction time. Similarly, default function arguments are evaluated when the function is called.

Reply:
Are you not worried about a function call on each construction here? I guess element segments are not that common...

Reply:
I am slightly worried about using the `Type` constructor in more places. It's rather expensive because it has to do a lookup in the global type store. I'll do some quick measurements to check that this isn't a big performance regression.

Reply:
Quick measurement didn't show anything concerning.